src/clang_parser.cpp

```diff
@@ -330,8 +330,7 @@ static SizedType get_sized_type(CXType clang_type)
       // Only support one-dimensional arrays for now
       if (elem_type.kind != CXType_ConstantArray)
       {
-        auto type = get_sized_type(elem_type);
-        return CreateArray(size, type.type, type.size, type.IsSigned());
+        return CreateArray(size, get_sized_type(elem_type));
```

Comment:
Is `size` correct? Shouldn't it be `size / <elem_size>`?

Reply:
I don't think so as size already was the number of elements, that bit hasn't changed.

Looking at the clang src itself this method is equal to `clang_getNumElements` for a constant array.
```
long long clang_getNumElements(CXType CT) {
  long long result = -1;
  QualType T = GetQualType(CT);
  const Type *TP = T.getTypePtrOrNull();

  if (TP) {
    switch (TP->getTypeClass()) {
    case Type::ConstantArray:
      result = cast<ConstantArrayType> (TP)->getSize().getSExtValue();
      break;
    case Type::Vector:
      result = cast<VectorType> (TP)->getNumElements();
      break;
    case Type::ExtVector:
      result = cast<ExtVectorType>(TP)->getNumElements();
      break;
    default:
      break;
    }
  }
  return result;
}

long long clang_getArraySize(CXType CT) {
  long long result = -1;
  QualType T = GetQualType(CT);
  const Type *TP = T.getTypePtrOrNull();

  if (TP) {
    switch (TP->getTypeClass()) {
    case Type::ConstantArray:
      result = cast<ConstantArrayType> (TP)->getSize().getSExtValue();
      break;
    default:
      break;
    }
  }
  return result;
}
```

I'll switch it to `clang_getNumElements` to make it less confusing